org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionResolveHandler.java

```diff
@@ -80,9 +80,11 @@
 	public static final String DEFAULT = "Default: ";
 	private static final String VALUE = "Value: ";
 	private final PreferenceManager manager;
+	private final ExecutorService executorService;
 
-	public CompletionResolveHandler(PreferenceManager manager) {
+	public CompletionResolveHandler(ExecutorService executorService, PreferenceManager manager) {
```

Comment:
I like the idea of reusing the thread pool. One thing with this PR is to keep it backward compatible. Currently the new constructor will break VS Code IntelliCode extension, which is calling the old constructor to instantize a CompletionResolveHandler.

Another idea is to move ExecutorService instance to BaseJDTLanguageServer, and other places can access it via getter like `JavaLanguageServerPlugin.getProtocol().getCachedThreadPool()`.  In this way, both syntax server and standard server share the same logic, and no API changes are required in Handlers.

Reply:
How about keeping both two implementation, so downstream extensions like IntelliCode won't break immediately. Later we can update them to call the new constructor.

```java
public CompletionResolveHandler(ExecutorService executorService, PreferenceManager manager);

@Deprecated
public CompletionResolveHandler(PreferenceManager manager);
```

Reply:
> How about keeping both two implementation, so downstream extensions like IntelliCode won't break immediately. Later we can update them to call the new constructor if necessary
> 
> ```java
> public CompletionResolveHandler(ExecutorService executorService, PreferenceManager manager);
> 
> public CompletionResolveHandler(PreferenceManager manager);
> ```

This may be an option, but it would require the downstream extensions to make the same changes again. If we could make this benefit available to downstream extensions for free, I would vote for it.

Reply:
> If we could make this benefit available to downstream extensions for free

Downstream extension E.g. IntelliCode is using `CompletionResolveHandler(PreferenceManager manager)`. Can we just change the implementation of it to use a shared ExecutorService?

Reply:
I had a quick go at this but unit tests fail because getProtocol returns null. 

Are there already tests in place that set that up so I could have a look at how it's done?

Reply:
You could take a loo at
https://github.com/eclipse/eclipse.jdt.ls/blob/346d6fa63d00f3a30091ac1984e1cd835b13e83f/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerTest.java#L521-L530